cl/sentinel/handlers/heartbeats.go

```diff
@@ -24,30 +24,50 @@ import (
 // Since packets are just structs, they can be resent with no issue
 
 func (c *ConsensusHandlers) pingHandler(s network.Stream) error {
+	peerId := s.Conn().RemotePeer().String()
+	if err := c.checkRateLimit(peerId, "ping", defaultRateLimits.pingLimit); err != nil {
+		return err
```

Comment:
if the rate limit check fails send write to the stream []byte{0x2} and close it, this is per spec the status code for rate limiting